### PR TITLE
TypeScript Styled Components, DownloadCSV and Tooltips

### PR DESCRIPTION
**Pull request by cazapatamar**

### CODE REVIEW

File: src/pages/search/shared_components/CssLegends.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+const ThickLineLegend = styled(LineLegend)<PropsThickLineLeg>`
+  &:before {
+    border-bottom: 8px solid ${(props) => props.color};
+    height: 0px;
+  }
+`;
+
+interface PropsTextLeg {
+  color: string;
+  image: string;
```
Comment (erikasv) [inline src/pages/search/shared_components/CssLegends.tsx:90]: according to the implementation, image should be optional

File: src/pages/search/shared_components/CssLegends.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  hoverImage: string;
```
Comment (erikasv) [inline src/pages/search/shared_components/CssLegends.tsx:91]: hoverImage too

File: src/pages/search/shared_components/CssLegends.tsx (hunk 1 above)
Comment (cazapatamar) [reply src/pages/search/shared_components/CssLegends.tsx]: Done!

File: src/pages/search/shared_components/CssLegends.tsx (hunk 2 above)
Comment (cazapatamar) [reply src/pages/search/shared_components/CssLegends.tsx]: Done!

**Review by erikasv — APPROVED** [review]

**Review by Trjegul84 — APPROVED** [review]

**Pull request merged**